solutions/USACO jan20-race.mdx

```diff
@@ -0,0 +1,115 @@
+---
+id: usaco-989
+source: USACO Bronze 2020 January
+title: Race
+author: Jesse Choe
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_race_bronze_jan20.html)
+
+```cpp
+// Created by Jesse Choe - Bronze Template
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// Type aliases
+
+using ll = long long;
+using str = string; 
+using vi = vector<int>;
+using pi = pair<int,int>;
+using vpi = vector<pi>;
+using si = set<int>;
+
+// Vector Operations
+
+#define sz(x) (int)(x).size()
+#define pb push_back
+#define all(x) begin(x), end(x)
+#define sor(x) sort(all(x))
+#define rev(x) reverse(all(x))
+#define del(x, i) (x).erase(begin(x)+i)
+#define rem(x,i) (x).erase(begin(x), begin(x)+i)
+
+// Pairs
+
+#define mp make_pair
+#define sec second
+#define fir first
+
+// Sets and Maps
+
+#define ins insert
+#define ez erase
+#define cnt count
+
+// Math
+
+const int MOD = 1e9+7;
+const int MAX_INT = 2147483647;
+const int MIN_INT = -2147483647;
+
+// Permutation
+
+#define possibilities(x) while(next_permutation(all(x))
+
+// Loops
+
+#define Trav(a,x) for (auto& a: x)
+#define For(i,a,b) for (int i = (a); i < (b); ++i)
+
+// Input/Output - s is basically the file name without the ".in" and ".out"
+
+void setIO(string name="") { // name is nonempty for USACO file I/O
+    ios_base::sync_with_stdio(0); cin.tie(0); 
+    if(sz(name)){
+        freopen((name+".in").c_str(), "r", stdin);
+        freopen((name+".out").c_str(), "w", stdout);
+    }
+}
+
+int main(){
+
+	setIO("race"); 
+
+	int k, N; cin >> k >> N;
+
+	int p = 1;
+	bool reached = false;
+	while(p*(p+1) <= 2*k){
+		p++;
+	}
+	p--;
+	if(2*k == p*(p+1)){
+		reached = true;
+	}
+
+	For(i, 0, N){
+		
+		int x; cin >> x;
+		
+		if(x >= p){
+			if(reached){
+				cout << p << endl;
+			} else {
+				cout << p+1 << endl;
+			}
+			continue;
+		}
+		
+		int y = sqrt(double(((2*k+pow(x, 2)-x) / 2)));
```

Comment:
don't need `double` on this line

Reply:
Oops, it seems like I put the wrong code after looking at it. MB. I have another version using simulation.